docs/core/docker/publish-as-container.md

```diff
@@ -0,0 +1,349 @@
+
+By default, the value used will be the `Version` of the project.
+
+```xml
+<ContainerImageTag>1.2.3-alpha2</ContainerImageTag>
+```
+
+```xml
+<ContainerImageTags>1.2.3-alpha2;latest</ContainerImageTags>
+```
+
+Tags can only contain up to 127 alphanumeric characters, periods, underscores, and dashes. They must start with an alphanumeric character or an underscore. Any other form will result in an error being thrown.
+
+### `ContainerWorkingDirectory`
+
+This property controls the working directory of the container - the directory that commands are executed within if not other command is run.
+
+By default, we use the `/app` directory as the working directory.
+
+```xml
+<ContainerWorkingDirectory>/bin</ContainerWorkingDirectory>
+```
+
+## `ContainerPort`
+
+This item adds TCP or UDP ports to the list of known ports for the container. This enables container runtimes like Docker to map these ports to the host machine automatically. This is often used as documentation for the container, but can also be used to enable automatic port mapping.
+
+ContainerPort items have two properties:
+
+* Include
+  * The port number to expose
+* Type
+  * One of `tcp` or `udp` - the default is `tcp`
+
+```xml
+<ItemGroup>
+    <ContainerPort Include="80" Type="tcp" />
+</ItemGroup>
+```
+
+> **Note**
+> This item does nothing for the container by default and should be considered advisory at best.
+
+### `ContainerLabel`
+
+This item adds a metadata label to the container. Labels have no impact on the container at runtime, but are often used to store version and authoring metadata for use by security scanners and other infrastructure tools.
+
+ContainerLabel items have two properties:
+
+* Include
+  * The key of the label
+* Value
+  * The value of the label - this may be empty
+
+See [default container labels](#default-container-labels) for a list of labels that are created by default.
+
+```xml
+<ItemGroup>
+    <ContainerLabel Include="org.contoso.businessunit" Value="contoso-university" />
+<ItemGroup>
+```
+
+## `ContainerEnvironmentVariable`
+
+This item adds a new environment variable to the container. Environment variables will be accessible to the application running in the container immediately, and are often used to change the runtime behavior of the running application.
+
+ContainerEnvironmentVariable items have two properties:
+
+* Include
+  * The name of the environment variable
+* Value
+  * The value of the environment variable
+
+```xml
+<ItemGroup>
+  <ContainerEnvironmentVariable Include="LOGGER_VERBOSITY" Value="Trace" />
+</ItemGroup>
+```
+
+## `ContainerEntrypoint`
+
+This item can be used to customize the entrypoint of the container - the binary that is run by default when the container is started.
+
+By default, for builds that create an executable binary that binary is set as the ContainerEntrypoint. For builds that do not create an executable binary `dotnet path/to/application.dll` is used as the ContainerEntrypoint.
```

Comment:
We often call this binary an `AppHost`. Have you seen this term used in other Learn content? Do you think users would recognize it instead of the phrase 'an executable binary'?

Reply:
Maybe "app host", I'll update this.